src/utils/historical-data-source.ts

```diff
@@ -0,0 +1,11 @@
+export class HistoricalDataSourceService {
```

Comment:
It feels wrong having a service live under "utils", since it's Datasource related, I would rather move this to the "datasource" folder.